### ISSUE TITLE
Segmentation fault on unit Test

### ISSUE THREAD

**Issue by Surt** [Codeception/Codeception#3781]
Yesterday, after upgrading codecept to the last version (last one I was using was 1 month older, don't know the version) my unit tests fails with a segmentation fault.

```
codecept run unit -vvv 
Codeception PHP Testing Framework v2.2.6
Powered by PHPUnit 5.6.3 by Sebastian Bergmann and contributors.

Unit Tests (60) --------------------------------------------------------------------------
Modules: Asserts, \Helper\Unit, Mockery, Phalcon
------------------------------------------------------------------------------------------
✔ AclTest: Acl acesses (0.03s)
✔ SearchServiceTest: Searcher (0.15s)
✔ SearchServiceTest: Searcher invalid search (0.01s)
✔ SearchTest: Default build (0.06s)
✔ SearchTest: Params build (0.06s)
✔ ControllerWebTest: Sanitize input_with corrupted data_returns filtered values (0.01s)
✔ RouterTest: Testfront end | #0 (0.01s)
✔ RouterTest: Testfront end | #1 (0.02s)
✔ RouterTest: Testfront end | #2 (0.01s)
✔ RouterTest: Testfront end | #3 (0.01s)
✔ RouterTest: Testfront end | #4 (0.01s)
✔ RouterTest: Testfront end | #5 (0.01s)
✔ RouterTest: Testback end | #0 (0.02s)
✔ RouterTest: Testback end | #1 (0.01s)
- RouterTest: Testback end | #2Segmentation fault
```

If I run the unit tests by hand, one by one, specifying `codecept run unit FolderHere` everything works fine.
Before the update there was no segmentation fault. I tried multiple things, like updating phpunit, xdebug extension, checking yml etc... but nothing works.
Any clue?

**Comment by Naktibalda** [comment]
Phalcon framework has its own C extention. Are you still using the same version?

Please try to get the core dump file and analyze it using [gdb](https://bugs.php.net/bugs-generating-backtrace.php).

**Comment by Surt** [comment]
yep, all the same, even the tests :( the only change was a codecept self-update. I will try to get the core dump file

**Comment by sergeyklay** [comment]
@Surt Could you please provide a script to reproduce?

**Comment by Surt** [comment]
@sergeyklay each unit test folder runs ok, the problem only comes when triying to run all of them at once.
There are many of them. I'm using the Phalcon Module, each "Cest" file does have a "_before", sometimes a "setUp" where I set some services on the Di, for example, the router one:
    
```php
protected function _before()
    {
        // Load any additional services that might be required during testing
        $router_path = CONFIG_PATH.'router.php';
        $router = include $router_path;

        $this->tester->haveServiceInDi("router", $router);
        $this->router = $this->tester->grabServiceFromDi('router');
    }
```

By the way, my unit cest files extends from `\Codeception\TestCase\Test` I don't know if this is the correct way.
I was unable to get the core dump, php-fpm, virtual servers, etc.. is for me a pain...

**Comment by Naktibalda** [comment]
php-fpm is irrelevant, because Codeception run in command line.

Try to run `strace codecept run unit -vvv`

**Comment by Surt** [comment]
Hi @Naktibalda ,
here it comes. [strace output](https://s3-eu-west-1.amazonaws.com/uploads-eu.hipchat.com/83000/2018209/IyQc4aBqvufwwCD/straceCodecept.txt)
No clue about the strace output, only the `--- SIGSEGV {si_signo=SIGSEGV, si_code=SEGV_MAPERR, si_addr=0x16fc} ---
+++ killed by SIGSEGV +++`

**Comment by Surt** [comment]
I tried deleting test by test, to search for adisruption point, but there was no look. It happends when I run all the unit Test at once.

below you can see, it breaks even when there is no problems on the tests.
```
codecept run unit Utilities
Codeception PHP Testing Framework v2.2.6
Powered by PHPUnit 5.6.3 by Sebastian Bergmann and contributors.

Unit Tests (35) ---------------------------------------------------------------------------------------------------------------------
✔ MoneyUtilitiesTest: Extract percentage (0.01s)
✔ MoneyUtilitiesTest: Testget symbol (0.01s)
✔ MoneyUtilitiesTest: Testget converted amount (0.01s)
✔ MoneyUtilitiesTest: Testto string (0.01s)
✔ MoneyUtilitiesTest: Testparse string (0.01s)
✔ InputTest: From bag get_with no params from bag with one value_returns correct input (0.04s)
✔ InputTest: From bag get_with no params from bag with more than one value_returns correct input (0.01s)
✔ InputTest: From bag get_with one param from bag with more than one value_returns correct input (0.01s)
✔ InputTest: From bag get_with multiple params from bag with more than one value_returns correct input (0.01s)
✔ InputTest: From bag get_with multiple non existing params from bag with more than one value_returns correct input (0.01s)
✔ InputTest: Store_with correct input bag and corrupted params_stores sanitized input (0.03s)
✔ InputTest: Store_with no params and single element input_stores correct input (0.01s)
✔ InputTest: Store_with no params and multiple element input_stores correct input (0.01s)
✔ InputTest: Store_with params_stores correct input (0.01s)
✔ InputTest: Store_with params_sanitizes input (0.01s)
✔ InputTest: Store_with a param with not correct sanitization filter_throws exception (0.01s)
✔ InputTest: Get_with correct input bag and corrupted params_gets sanitized input (0.01s)
✔ InputTest: Testget_with param with wrong sanitization filters_throws exception | #0 (0.01s)
✔ InputTest: Testget_with param with wrong sanitization filters_throws exception | #1 (0.02s)
✔ InputTest: Testget_with param with wrong sanitization filters_throws exception | #2 (0.01s)
✔ InputTest: Testget_with param with wrong sanitization filters_throws exception | #3 (0.01s)
✔ InputTest: Testget_with param with wrong sanitization filters_throws exception | #4 (0.04s)
✔ InputTest: Get_with one param_returns sanitized subset of input (0.02s)
✔ InputTest: Get_with multiple_returns sanitized subset of input (0.01s)
✔ InputTest: Get_with encoded params_returns sanitized and correct input (0.02s)
✔ InputTest: Store_with param with no sanitization filters_throws exception | #0 (0.02s)
✔ InputTest: Store_with param with no sanitization filters_throws exception | #1 (0.01s)
✔ InputTest: Store_with param with no sanitization filters_throws exception | #2 (0.02s)
✔ InputTest: Store_with param with no sanitization filters_throws exception | #3 (0.02s)
✔ InputTest: Store_with param with no sanitization filters_throws exception | #4 (0.01s)
✔ InputTest: Store_with no params and not processed input_stores null values (0.02s)
✔ InputTest: Delete_with filled persistor_deletes correctly (0.01s)
✔ InputTest: Get raw input_with no source_throws exception (0.01s)
✔ InputTest: Sanitize input_with corrupted params_sanitizes input (0.01s)
✔ InputTest: Test_with bad input bag name_throws exception (0.01s)
-------------------------------------------------------------------------------------------------------------------------------------


Time: 1.54 seconds, Memory: 27.00MB

OK (35 tests, 38 assertions)


codecept run unit 
Codeception PHP Testing Framework v2.2.6
Powered by PHPUnit 5.6.3 by Sebastian Bergmann and contributors.

Unit Tests (47) ---------------------------------------------------------------------------------------------------------------------
✔ AclTest: Acl acesses (0.03s)
✔ SearchServiceTest: Searcher (0.10s)
✔ SearchServiceTest: Searcher invalid search (0.01s)
✔ SearchTest: Default build (0.06s)
✔ SearchTest: Params build (0.06s)
✔ ControllerWebTest: Sanitize input_with corrupted data_returns filtered values (0.01s)
✔ InvoiceServiceTest: Invoice build (0.08s)
✔ RequestConfigBuilderTest: Get uri scheme_is dev environment_returns http (0.02s)
✔ RequestConfigBuilderTest: Get uri scheme_is not dev environment_returns https (0.01s)
✔ RequestConfigBuilderTest: Get uri host_is not dev environment_returns prod (0.01s)
✔ RequestConfigBuilderTest: Get uri host_is dev environment_returns dev (0.01s)
✔ ProviderServiceTest: Get by name_with empty search_throws exception (0.01s)
✔ MoneyUtilitiesTest: Extract percentage (0.01s)
✔ MoneyUtilitiesTest: Testget symbol (0.01s)
✔ MoneyUtilitiesTest: Testget converted amount (0.01s)
✔ MoneyUtilitiesTest: Testto string (0.01s)
✔ MoneyUtilitiesTest: Testparse string (0.01s)
✔ InputTest: From bag get_with no params from bag with one value_returns correct input (0.02s)
✔ InputTest: From bag get_with no params from bag with more than one value_returns correct input (0.01s)
✔ InputTest: From bag get_with one param from bag with more than one value_returns correct input (0.01s)
✔ InputTest: From bag get_with multiple params from bag with more than one value_returns correct input (0.01s)
- InputTest: From bag get_with multiple non existing params from bag with more than one value_returns correct inputSegmentation fault

```

**Comment by Naktibalda** [comment]
We can't help if you can't provide the backtrace from core dump.

Try this way:
command: `gdb /usr/bin/php`
enter:  `run vendor/bin/codecept run unit`
if it crashed, enter: `bt`

**Comment by Surt** [comment]
```
gdb /usr/bin/php
GNU gdb (Debian 7.7.1+dfsg-5) 7.7.1
Copyright (C) 2014 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.  Type "show copying"
and "show warranty" for details.
This GDB was configured as "x86_64-linux-gnu".
Type "show configuration" for configuration details.
For bug reporting instructions, please see:
<http://www.gnu.org/software/gdb/bugs/>.
Find the GDB manual and other documentation resources online at:
<http://www.gnu.org/software/gdb/documentation/>.
For help, type "help".
Type "apropos word" to search for commands related to "word"...
Reading symbols from /usr/bin/php...(no debugging symbols found)...done.
(gdb) run /usr/local/bin/codecept run unit
Starting program: /usr/bin/php /usr/local/bin/codecept run unit
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/lib/x86_64-linux-gnu/libthread_db.so.1".
[New Thread 0x7fffe8297700 (LWP 11811)]
[Thread 0x7fffe8297700 (LWP 11811) exited]
Codeception PHP Testing Framework v2.2.6
Powered by PHPUnit 5.6.3 by Sebastian Bergmann and contributors.

Unit Tests (60) ---------------------------------------------------------------------------------------------------------------------
✔ AclTest: Acl acesses (0.02s)
✔ SearchServiceTest: Searcher (0.10s)
✔ SearchServiceTest: Searcher invalid search (0.01s)
✔ SearchTest: Default build (0.06s)
✔ SearchTest: Params build (0.06s)
✔ ControllerWebTest: Sanitize input_with corrupted data_returns filtered values (0.01s)
✔ RouterTest: Testfront end | #0 (0.02s)
✔ RouterTest: Testfront end | #1 (0.02s)
✔ RouterTest: Testfront end | #2 (0.02s)
✔ RouterTest: Testfront end | #3 (0.01s)
✔ RouterTest: Testfront end | #4 (0.01s)
✔ RouterTest: Testfront end | #5 (0.01s)
✔ RouterTest: Testback end | #0 (0.01s)
✔ RouterTest: Testback end | #1 (0.02s)
✔ RouterTest: Testback end | #2 (0.01s)
✔ RouterTest: Testback end | #3 (0.01s)
✔ RouterTest: Testback end | #4 (0.01s)
✔ RouterTest: Testback end | #5 (0.01s)
✔ RouterTest: Api | #0 (0.01s)
✔ InvoiceServiceTest: Invoice build (0.10s)
✔ RequestConfigBuilderTest: Get uri scheme_is dev environment_returns http (0.04s)
✔ RequestConfigBuilderTest: Get uri scheme_is not dev environment_returns https (0.01s)
- RequestConfigBuilderTest: Get uri host_is not dev environment_returns prod
Program received signal SIGSEGV, Segmentation fault.
0x0000000000714ae8 in zend_std_object_get_class ()
```

**Comment by Naktibalda** [comment]
It looks like PHP bug [#72286](https://bugs.php.net/bug.php?id=72286).

**Comment by Surt** [comment]
:( well, I will follow the bug until its resolved and post here the results. 
Thank you very much for your efforts !!

**Comment by Naktibalda** [comment]
Going back to your first post - do your tests pass if you downgrade Codeception?

**Comment by Surt** [comment]
:( no, it does not work with downgrades of codecept now. Same behaviour. even using different enviroment servers.
repository changes show no changes on tests... :( I will drop it for the moment, too much time on this

Thank you so much.

**Comment by Naktibalda** [comment]
OK, I am closing this ticket, because it is an issue of PHP, not of Codeception.

**Comment by Naktibalda** [comment]
@Surt It is unlikely that this PHP issue will be fixed, active support of PHP 5.6 ends at the end of this month.
